From: Jan Grulich <jgrulich@redhat.com>
Date: Wed, 6 Mar 2024 10:19:00 +0000
Subject: Bug 1882438 - WebRTC backport: PipeWire camera - use length of device
 id instead display name r=pehrsons,webrtc-reviewers

This is a simple backport of an WebRTC upstream change.

Upstream commit: 16ac10d9f75cde959f00df062f544c49941882da

Differential Revision: https://phabricator.services.mozilla.com/D203099
Mercurial Revision: https://hg.mozilla.org/mozilla-central/rev/b572eb8b39b03f714234afff4bd80b4612439521
---
 modules/video_capture/linux/device_info_pipewire.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules/video_capture/linux/device_info_pipewire.cc b/modules/video_capture/linux/device_info_pipewire.cc
index f9f08a9c27..31d922035b 100644
--- a/modules/video_capture/linux/device_info_pipewire.cc
+++ b/modules/video_capture/linux/device_info_pipewire.cc
@@ -96,7 +96,7 @@ int32_t DeviceInfoPipeWire::CreateCapabilityMap(
       continue;
 
     _captureCapabilities = node.capabilities();
-    _lastUsedDeviceNameLength = node.display_name().length();
+    _lastUsedDeviceNameLength = node.unique_id().length();
     _lastUsedDeviceName = static_cast<char*>(
         realloc(_lastUsedDeviceName, _lastUsedDeviceNameLength + 1));
     memcpy(_lastUsedDeviceName, deviceUniqueIdUTF8,
